src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs

```diff
@@ -659,6 +647,18 @@ public static void Run(ServiceBase[] services)
             }
             finally
             {
+                // Free the pointer to the name of the service on the unmanaged heap.
+                // We don't need to free the last element in the unmanaged array since the last element have null values (denotes the end of the table).
+                for (int i = 0; i < services.Length; i++)
```

Comment:
If something failed before the entriesPointer got fully initialize, this is going to access random memory. This should only go over the memory that was initialized.

Reply:
Ah, true.